recipes/date/all/conanfile.py

```diff
@@ -1,29 +1,84 @@
 import os
-from conans import ConanFile, tools
+from conans import ConanFile, CMake, tools
 
 
 class DateConan(ConanFile):
     name = "date"
     url = "https://github.com/conan-io/conan-center-index"
     homepage = "https://github.com/HowardHinnant/date"
     description = "A date and time library based on the C++11/14/17 <chrono> header"
-    topics = ("date", "datetime", "timezone", "calendar", "time", "iana-database")
+    topics = ("date", "datetime", "timezone",
+              "calendar", "time", "iana-database")
     license = "MIT"
-    exports_sources = ["patches/*"]
-    no_copy_source = True
+    exports_sources = ["patches/*", "CMakeLists.txt"]
+    settings = "os", "arch", "compiler", "build_type"
+    generators = "cmake", "cmake_find_package"
+    options = {"shared": [True, False],
+               "fPIC": [True, False],
+               "use_system_tz_db": [True, False],
+               "use_tz_db_in_dot": [True, False]}
+    default_options = {"shared": False,
+                       "fPIC": True,
+                       "use_system_tz_db": True,
+                       "use_tz_db_in_dot": False}
+    _source_subfolder = "source_subfolder"
+    _build_subfolder = "build_subfolder"
```

Comment:
```suggestion
    @property
    def _source_subfolder(self):
        return "source_subfolder"

    @property
    def _build_subfolder(self):
        return "build_subfolder"
```

Reply:
Applied. yea - useful to achieve immutability.